backend/modules/embedder/embedder.py

```diff
@@ -0,0 +1,72 @@
+# A global registry to store all available embedders.
+from langchain.embeddings import CacheBackedEmbeddings
+from langchain.embeddings.base import Embeddings
+from langchain_community.storage.redis import RedisStore
+
+from backend.settings import settings
+from backend.types import EmbedderConfig, EmbeddingCacheConfig
+
+EMBEDDER_REGISTRY = {}
+
+
+def register_embedder(provider, cls):
+    """
+    Registers all the available loaders using `BaseEmbedder` class
+
+    Args:
+        cls: The embedder class to be registered.
+
+    Returns:
+        None
+    """
+    global EMBEDDER_REGISTRY
+    # Validate and add the embedder to the registry.
+    if not provider:
+        raise ValueError(
+            f"static attribute `name` needs to be a non-empty string on class {cls.__name__}"
+        )
+    if provider in EMBEDDER_REGISTRY:
+        raise ValueError(
+            f"Error while registering class {cls.__name__}, `name` already taken by {EMBEDDER_REGISTRY[provider].__name__}"
+        )
+    EMBEDDER_REGISTRY[provider] = cls
+
+
+def get_embedding_cache_store(config: EmbeddingCacheConfig):
+    if config.provider == "redis":
+        return RedisStore(
+            redis_url=config.url,
+            client_kwargs=config.config,
+            namespace="embedding_caches",
+        )
+    raise Exception(f"Embedding cache provider {config.provider} not supported!")
+
+
+def get_embedder(
+    embedder_config: EmbedderConfig,
+) -> Embeddings:
```

Comment:
Currently the base class is tied to langchain
We will define an agnostic interface/protocol in future